tms/__main__.py

```diff
@@ -0,0 +1,144 @@
+                **args,  # TODO
+            )
+            # confirm start (otherwise ewms will request this one again -- good for statelessness)
+            await ewms_rc.request(
+                "PATCH",
+                f"/tms/taskforce/start/{args['taskforce_uuid']}",
+                ewms_taskforce_attrs,
+            )
+            LOGGER.info("Sent taskforce info to EWMS")
+
+        await asyncio.sleep(ENV.TMS_OUTER_LOOP_WAIT)
+
+
+async def watcher_loop() -> None:
+    """Watch over all job event log files and send EWMS taskforce updates."""
+    in_progress: dict[Path, asyncio.Task[None]] = {}
+
+    # make connections -- do now so we don't have any surprises downstream
+    ewms_rc = RestClient(ENV.EWMS_ADDRESS, token=ENV.EWMS_AUTH)
+    LOGGER.info("Connected to EWMS")
+
+    while True:
+        for jel_fpath in ENV.JOB_EVENT_LOG_DIR.iterdir():
+            if jel_fpath in in_progress:
+                continue
+            task = asyncio.create_task(watcher.watch_job_event_log(jel_fpath, ewms_rc))
+            await asyncio.sleep(0)  # start above task
+            in_progress[jel_fpath] = task
+
+        await asyncio.sleep(ENV.TMS_OUTER_LOOP_WAIT)
+
+
+async def stopper_loop() -> None:
+    """Listen to EWMS and stop designated taskforces."""
+
+    # make connections -- do now so we don't have any surprises downstream
+    schedd_obj = htcondor.Schedd()  # no auth need b/c we're on AP
+    ewms_rc = RestClient(ENV.EWMS_ADDRESS, token=ENV.EWMS_AUTH)
+    LOGGER.info("Connected to EWMS")
+
+    async def next_one() -> dict[str, Any]:
+        """Get the next taskforce requested for this collector + schedd."""
+        return await ewms_rc.request(  # type: ignore[no-any-return]
+            "GET",
+            "/tms/taskforce/stop",
+            {"collector": ENV.COLLECTOR, "schedd": ENV.SCHEDD},
+        )
+
+    while True:
+        while args := await next_one():
+            stopper.stop(
+                schedd_obj,
+                args["cluster_id"],
+            )
+            # confirm stop (otherwise ewms will request this one again -- good for statelessness)
+            await ewms_rc.request(
+                "DELETE",
+                f"/tms/taskforce/stop/{args['taskforce_uuid']}",
+            )
+
+        await asyncio.sleep(ENV.TMS_OUTER_LOOP_WAIT)
+
+
+def _create_loop_task(key: str) -> asyncio.Task[None]:
+    funcs = {
+        "starter": starter_loop,
+        "watcher": watcher_loop,
+        "stopper": stopper_loop,
+    }
+    return asyncio.create_task(funcs[key]())
+
+
+async def main() -> None:
+    """explain."""
+    htcondor.set_subsystem("TOOL")
+    htcondor.param["TOOL_DEBUG"] = "D_FULLDEBUG"
+    # htcondor.param["TOOL_LOG"] = "log.txt"
+    # htcondor.enable_log()
+    htcondor.enable_debug()
+
+    loops = {
+        k: _create_loop_task(k)
+        for k in [
+            "starter",
+            "watcher",
+            "stopper",
+        ]
+    }
```

Comment:
This seems overly complicated to call a function that looks up a key in a dict to create a task?  Not sure why this can't be written more clearly like 
```python
loops = {
    "starter": asyncio.create_task(starter_loop()),
    "watcher": asyncio.create_task(watcher_loop()),
    "stopper": asyncio.create_task(stopper_loop()),
}
```

As for restarting down below, you shouldn't ever have to do that?  They should only exit with an exception, which hopefully you're catching inside, or it's important enough to exit completely.

Reply:
I greatly simplified this. Should be better now